.github/workflows/ci.yml

```diff
@@ -712,3 +725,166 @@ jobs:
+    if: ${{ github.repository == 'silverstripe/silverstripe-framework' && github.repository_owner == 'silverstripe' && (github.event_name == 'push' ||  github.event_name == 'schedule') }}
+    env:
+      GITHUB_REPOSITORY: ${{ github.repository }}
+      GITHUB_REF_NAME: ${{ github.ref_name }}
+      GITHUB_SHA: ${{ github.sha }}
+      LATEST_LOCAL_SHA: ${{ needs.tests.outputs.latest_local_sha }}
+    outputs:
+      do_release: ${{ steps.determine-if.outputs.do_release }}
+      release_notes: ${{ steps.determine-if.outputs.release_notes }}
+      latest_tag: ${{ steps.determine-if.outputs.latest_tag }}
+      next_tag: ${{ steps.determine-if.outputs.next_tag }}
+    steps:
+
+      - name: Determine if should patch release
+        id: determine-if
+        run: |
+          DO_RELEASE=1
+
+          # Must be on a minor branch to do a patch release
+          if ! [[ $GITHUB_REF_NAME =~ ^[0-9]+\.[0.9]+$ ]]; then
+            echo "Not patch releasing because not on a minor branch"
+            DO_RELEASE=0
+          fi
+
+          # Validate that this commit is that latest commit for the branch using GitHub API
+          # We need to check this in case re-rerunning an old job and there have been new commits since
+          if [[ $DO_RELEASE == "1" ]]; then
+            # https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28
+            AMP='&'
+            RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+            -X GET https://api.github.com/repos/${GITHUB_REPOSITORY}/commits?sha=${GITHUB_REF_NAME}${AMP}per_page=1 \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            )
+            if [[ $RESP_CODE != "200" ]]; then
+              echo "Unable to read list of commits - HTTP response code was $RESP_CODE"
+              exit 1
+            fi
+            LATEST_REMOTE_SHA=$(jq -r '.[0].sha' __response.json)
+            echo "LATEST_REMOTE_SHA is $LATEST_REMOTE_SHA"
+            echo "LATEST_LOCAL_SHA is $LATEST_LOCAL_SHA"
+            if [[ $LATEST_REMOTE_SHA != $LATEST_LOCAL_SHA ]]; then
+              echo "Not patch releasing because latest remote sha is not equal to latest local sha"
+              DO_RELEASE=0
+            fi
+            # Also validate the sha matches GITHUB_SHA, which is what gha-tag-release will use
+            if [[ $GITHUB_SHA != $LATEST_LOCAL_SHA ]]; then
+              echo "Not patch releasing because GITHUB_SHA is not equal to latest local sha"
+              DO_RELEASE=0
+            fi
+          fi
+
+          # Check is there is an existing tag on the branch using GitHub API
+          # Note cannot use local `git tag` because actions/checkout by default will not checkout tags
+          # and you need to checkout full history in order to get them
+          LATEST_TAG=""
+          NEXT_TAG=""
+          if [[ $DO_RELEASE == "1" ]]; then
```

Comment:
Given the exit above, we don't need this condition anymore.

Reply:
I'd prefer to keep the variable rather than switch to a function